javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1131 @@
+      protected def valueClazz: Class[V] = valueType
+      protected val inner: SortedMap[K, V] = m
+    }
+    m match {
+      case _: Serializable => new BasicCheckedSortedMap with Serializable
+      case _               => new BasicCheckedSortedMap
+    }
+  }
+
+  def emptyIterator[T](): Iterator[T] = {
+    new Iterator[T] {
+      override def hasNext: Boolean = false
+
+      override def next(): T =
+        throw new NoSuchElementException
+
+      override def remove(): Unit =
+        throw new IllegalStateException
+    }
+  }
+
+  def emptyListIterator[T](): ListIterator[T] = {
+    new ListIterator[T] {
+      override def hasNext: Boolean = false
+
+      override def next(): T =
+        throw new NoSuchElementException
+
+      override def hasPrevious: Boolean = false
+
+      override def previous(): T =
+        throw new NoSuchElementException
+
+      override def nextIndex(): Int = 0
+
+      override def previousIndex(): Int = -1
+
+      override def remove(): Unit =
+        throw new IllegalStateException
+
+      override def set(e: T): Unit =
+        throw new IllegalStateException
+
+      override def add(e: T): Unit =
+        throw new UnsupportedOperationException
+    }
+  }
+
+  def emptyEnumeration[T](): Enumeration[T] = {
+    new Enumeration[T] {
+      def hasMoreElements: Boolean = false
+
+      def nextElement(): T =
+        throw new NoSuchElementException
+    }
+  }
+
+  final def emptySet[T](): Set[T] =
+    EMPTY_SET.asInstanceOf[Set[T]]
+
+  final def emptyList[T](): List[T] =
+    EMPTY_LIST.asInstanceOf[List[T]]
+
+  final def emptyMap[K, V](): Map[K, V] =
+    EMPTY_MAP.asInstanceOf[Map[K, V]]
+
+  def singleton[T](o: T): Set[T] = {
+    unmodifiableSet(new AbstractSet[T] with io.Serializable {
+      def size(): Int = 1
+
+      def iterator(): Iterator[T] = {
+        new Iterator[T] {
+          private var _hasNext: Boolean = true
+
+          def hasNext: Boolean = _hasNext
+
+          def next(): T = {
+            if (!_hasNext)
+              throw new IllegalStateException
+            _hasNext = false
+            o
+          }
+
+          def remove(): Unit =
+            throw new UnsupportedOperationException
+        }
+      }
+    })
+  }
+
+  def singletonList[T](o: T): List[T] = {
+    unmodifiableList(new AbstractList[T] with io.Serializable {
+      def size(): Int = 1
+
+      def get(index: Int): T =
+        if (index === 0) o
+        else throw new IndexOutOfBoundsException(index.toString)
+    })
+  }
+
+  def singletonMap[K, V](key: K, value: V): Map[K, V] = {
+    unmodifiableMap[K, V](new AbstractMap[K, V] with io.Serializable {
+      def entrySet(): Set[Map.Entry[K, V]] =
+        singleton(new AbstractMap.SimpleImmutableEntry(key, value))
+    })
+  }
+
+  def nCopies[T](n: Int, o: T): List[T] = {
+    if (n < 0)
+      throw new IllegalArgumentException
+
+    unmodifiableList(new AbstractList[T] with RandomAccess {
```

Comment:
I've also added `unmodifiableList ` for to fail fast in case of attempt to modify the list.